homeassistant/components/vallox/__init__.py

```diff
@@ -170,44 +191,33 @@ async def async_update_data() -> ValloxState:
             schema=service_details.schema,
         )
 
-    hass.data[DOMAIN] = {"client": client, "coordinator": coordinator, "name": name}
-
-    async def _async_load_platform_delayed(*_: Any) -> None:
-        # We need a successful update before loading the platforms, because platform init code
-        # derives the UUIDs from the data the coordinator fetches.
-        warned_once = False
-        while hass.state == CoreState.running:
-            await coordinator.async_refresh()
-            if coordinator.last_update_success:
-                break
-
-            if not warned_once:
-                _LOGGER.warning(
-                    "Vallox integration not ready yet; Retrying in background"
-                )
-                warned_once = True
-
-            await asyncio.sleep(INITIAL_COORDINATOR_UPDATE_RETRY_INTERVAL_SECONDS)
-        else:
-            return
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
Not connected to this PR, but maybe we can fix it while we are here:

If we move the `async_config_entry_first_refresh()` to line 184 before registering the services, we can ensure that we don't offer services before we connected to the Vallox device safely at least once.

Reply:
Makes sense, fixed. 👍